lib/RPM/Grill.pm

```diff
@@ -940,10 +968,18 @@ sub gripe {
         }
     }
 
+    my $sig = gripe_signature(\%actual_gripe);
+
+    # Check if test case is not blacklisted, if it is, do not gripe about it.
+    $sig =~ /.*code:([a-zA-Z]+)\].*$/;
+    my $test_case = $1;
```

Comment:
I would convert that into a `sub is_blacklisted `
so that is can be tested in isolation and make the intend clear(er)